src/Symfony/Bridge/Twig/AppVariable.php

```diff
@@ -57,13 +72,7 @@ public function setDebug($debug)
      */
     public function getUser()
     {
-        if (null !== $this->tokenStorage) {
```

Comment:
Ah, I see what you mean now. Indeed the message will not be specific to the user anymore (will be about token). I'd indeed keep the check and throw an exception like before.